.github/workflows/tests-macos.yml

```diff
@@ -0,0 +1,212 @@
+      CMAKE_BUILD_PARALLEL_LEVEL: 4
+      PKG_CONFIG_PATH: /opt/homebrew/opt/lapack/lib/pkgconfig
+      LAPACKE_DIR: /opt/homebrew/opt/lapack
+      CC: ${{matrix.backend.c_compiler}}
+      CXX: ${{matrix.backend.cxx_compiler}}
+      CMAKE_BUILD_TYPE: ${{matrix.cmake_build_type}}
+    steps:
+    - name: Checkout built branch
+      uses: actions/checkout@v4
+      with: { submodules: true }
+    - name: Install PDI+user code plugin and dependencies
+      run: |
+        git clone --branch 1.8.1 --depth 1 https://github.com/pdidev/pdi.git
+        # PATCH: remove <link.h> for macOS
+        sed -i.bak 's|#include <link.h>||g' pdi/plugins/user_code/user_code.cxx
+        rm -f pdi/plugins/user_code/user_code.cxx.bak
+        cmake \
+          -DBUILD_BENCHMARKING=OFF \
+          -DBUILD_DECL_HDF5_PLUGIN=OFF \
+          -DBUILD_DECL_NETCDF_PLUGIN=OFF \
+          -DBUILD_DEISA_PLUGIN=OFF \
+          -DBUILD_DOCUMENTATION=OFF \
+          -DBUILD_FORTRAN=OFF \
+          -DBUILD_MPI_PLUGIN=OFF \
+          -DBUILD_PYCALL_PLUGIN=OFF \
+          -DBUILD_SERIALIZE_PLUGIN=OFF \
+          -DBUILD_SET_VALUE_PLUGIN=OFF \
+          -DBUILD_TESTING=OFF \
+          -DBUILD_TRACE_PLUGIN=OFF \
+          -DBUILD_USER_CODE_PLUGIN=ON \
+          -DCMAKE_CXX_FLAGS="-Wno-unqualified-std-cast-call" \
+          -DCMAKE_CXX_STANDARD=${{matrix.cxx_version}} \
+          -B build \
+          -S ./pdi
+        cmake --build build
+        cmake --install build --prefix $PDI_ROOT
+        rm -rf build
+    - name: Install fftw
+      run: brew install fftw
+    - name: Install lapack
+      run: brew install lapack
+    - run: git config --global --add safe.directory '*'
+    - name: Install Ginkgo
+      run: |
+        git clone --branch v1.8.0 --depth 1 https://github.com/ginkgo-project/ginkgo.git
+        cmake \
+          -DCMAKE_CXX_STANDARD=${{matrix.cxx_version}} \
+          -DGINKGO_BUILD_BENCHMARKS=OFF \
+          -DGINKGO_BUILD_EXAMPLES=OFF \
+          -DGINKGO_BUILD_MPI=OFF \
+          -DGINKGO_BUILD_REFERENCE=ON \
+          -DGINKGO_BUILD_TESTS=OFF \
+          -B build \
+          -S ./ginkgo
+        cmake --build build
+        cmake --install build --prefix $Ginkgo_ROOT
+        rm -rf build
+    - name: Install Google benchmark
+      run: |
+        cmake \
+          -DCMAKE_CXX_STANDARD=${{matrix.cxx_version}} \
+          -DBENCHMARK_ENABLE_GTEST_TESTS=OFF \
+          -DBENCHMARK_ENABLE_TESTING=OFF \
+          -DBENCHMARK_INSTALL_DOCS=OFF \
+          -DBENCHMARK_USE_BUNDLED_GTEST=OFF \
+          -B build \
+          -S ./vendor/benchmark
+        cmake --build build
+        cmake --install build --prefix $benchmark_ROOT
+        rm -rf build
+    - name: Install Google Test
+      run: |
+        cmake \
+          -DCMAKE_CXX_STANDARD=${{matrix.cxx_version}} \
+          -B build \
+          -S ./vendor/googletest
+        cmake --build build
+        cmake --install build --prefix $GTest_ROOT
+        rm -rf build
```

Comment:
Can you just use `brew install googletest`?
https://formulae.brew.sh/formula/googletest

Reply:
Yes I will give it a try

Reply:
Trying also with google benchmark

Reply:
done in 685625d68894d91217b58bfe9bbc785a704362b1